app/src/main/java/xyz/whereuat/whereuat/ui/views/AddContactDialogFragment.java

```diff
@@ -0,0 +1,119 @@
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import xyz.whereuat.whereuat.AsyncExecutor;
+import xyz.whereuat.whereuat.Constants;
+import xyz.whereuat.whereuat.MainActivity;
+import xyz.whereuat.whereuat.PendingRequestsActivity;
+import xyz.whereuat.whereuat.R;
+import xyz.whereuat.whereuat.utils.ContactUtils;
+import xyz.whereuat.whereuat.utils.PendingRequestUtils;
+
+/**
+ * This class is used for handling the dialog that is displayed when the user tries to add pending
+ * request for a number that was not found in the phonebook.
+ */
+public class AddContactDialogFragment extends DialogFragment {
+    public static final String TAG = "AddContactDialogFrag";
+
+    @Override
+    public Dialog onCreateDialog(final Bundle saved_instance_state) {
+        final Activity activity = getActivity();
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity, R.style.DialogTheme);
+        LayoutInflater inflater = activity.getLayoutInflater();
+        View view = inflater.inflate(R.layout.add_contact_dialog, (ViewGroup) this.getView());
+        builder.setView(view)
+                .setTitle(R.string.add_contact_prompt_title_text)
+                // When clicked, the "Okay" button should add the key location to the database if it
+                // is valid.
+                .setPositiveButton(R.string.okay, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Dialog d = (Dialog) dialog;
+                        EditText e = (EditText) d.findViewById(R.id.contact_name_input);
+
+                        String name = e.getText().toString();
+
+                        // Make sure the name is not null or empty and the location is not null.
+                        if (!name.isEmpty()) {
+                            addContact(activity, name, getArguments().getString(
+                                    Constants.CONTACT_NUMBER_BUNDLE_EXTRA));
+                        } else {
+                            String err_str = "Invalid contact name";
+                            Toast.makeText(activity, err_str, Toast.LENGTH_SHORT).show();
+                        }
+                    }
+                })
+                // The "Cancel" button shouldn't do anything.
+                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {}
+                });
+        return builder.create();
+    }
+
+    @Override
+    public void onStart() {
```

Comment:
Can we move this method? It's kind of just hanging out awkwardly between two closely-linked methods